Synthesis failed for AndroidPublisher

Hello! Autosynth couldn't regenerate AndroidPublisher. :broken_heart:

Here's the output from running `synth.py`:

```
2020-07-02 05:13:21,959 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-02 05:13:22,796 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-02 05:13:22,799 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-02 05:13:22,802 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-02 05:13:22,805 autosynth [DEBUG] > Running: git config push.default simple
2020-07-02 05:13:22,808 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-02 05:13:22,812 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-02 05:13:23,189 autosynth [INFO] > Running synthtool
2020-07-02 05:13:23,189 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-02 05:13:23,189 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-02 05:13:23,192 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-02 05:13:23,425 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-02 05:13:25,472 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpgdm2yt6d/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpgdm2yt6d/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:13:59.502 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-02 05:14:00,044 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpgdm2yt6d/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpgdm2yt6d/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-02 05:14:00,429 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpgdm2yt6d/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-02 05:14:00,461 autosynth [ERROR] > Synthesis failed
2020-07-02 05:14:00,461 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/939453dc-37db-4629-a9f2-9c9c07cee260/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-03 05:09:49,354 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-03 05:09:49,803 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-03 05:09:49,806 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-03 05:09:49,809 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-03 05:09:49,811 autosynth [DEBUG] > Running: git config push.default simple
2020-07-03 05:09:49,814 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-03 05:09:49,816 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-03 05:09:50,215 autosynth [INFO] > Running synthtool
2020-07-03 05:09:50,215 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-03 05:09:50,215 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-03 05:09:50,217 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-03 05:09:50,394 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-03 05:09:51,662 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpm51erp8t/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpm51erp8t/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:10:16.614 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-03 05:10:17,113 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpm51erp8t/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpm51erp8t/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-03 05:10:17,387 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpm51erp8t/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-03 05:10:17,412 autosynth [ERROR] > Synthesis failed
2020-07-03 05:10:17,413 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/16821607-0587-421f-acc3-066e6262d6ec/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-04 05:12:02,789 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-04 05:12:03,621 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-04 05:12:03,624 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-04 05:12:03,628 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-04 05:12:03,631 autosynth [DEBUG] > Running: git config push.default simple
2020-07-04 05:12:03,635 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-04 05:12:03,639 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-04 05:12:03,904 autosynth [INFO] > Running synthtool
2020-07-04 05:12:03,905 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-04 05:12:03,905 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-04 05:12:03,907 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-04 05:12:04,117 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-04 05:12:05,575 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpl7hf5oc6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpl7hf5oc6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:38.314 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-04 05:12:38,875 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpl7hf5oc6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpl7hf5oc6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-04 05:12:39,205 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpl7hf5oc6/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-04 05:12:39,236 autosynth [ERROR] > Synthesis failed
2020-07-04 05:12:39,236 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/008dd9d3-c752-43a8-a876-d8f4d739576a/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-05 05:11:40,099 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-05 05:11:40,917 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-05 05:11:40,920 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-05 05:11:40,923 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-05 05:11:40,925 autosynth [DEBUG] > Running: git config push.default simple
2020-07-05 05:11:40,928 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-05 05:11:40,931 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-05 05:11:41,165 autosynth [INFO] > Running synthtool
2020-07-05 05:11:41,165 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-05 05:11:41,165 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-05 05:11:41,167 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-05 05:11:41,380 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-05 05:11:43,165 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp1uqp0pcr/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp1uqp0pcr/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:12.807 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-05 05:12:13,535 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp1uqp0pcr/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp1uqp0pcr/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-05 05:12:13,895 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp1uqp0pcr/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-05 05:12:13,921 autosynth [ERROR] > Synthesis failed
2020-07-05 05:12:13,922 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/c74795ab-771b-4d59-a871-061dcb4bf3b8/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-06 05:12:04,214 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-06 05:12:05,042 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-06 05:12:05,045 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-06 05:12:05,047 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-06 05:12:05,050 autosynth [DEBUG] > Running: git config push.default simple
2020-07-06 05:12:05,053 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-06 05:12:05,057 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-06 05:12:05,326 autosynth [INFO] > Running synthtool
2020-07-06 05:12:05,326 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-06 05:12:05,326 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-06 05:12:05,328 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-06 05:12:05,554 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-06 05:12:07,443 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp4nyhzijn/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp4nyhzijn/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:38.646 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-06 05:12:39,222 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp4nyhzijn/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp4nyhzijn/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-06 05:12:39,561 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp4nyhzijn/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-06 05:12:39,587 autosynth [ERROR] > Synthesis failed
2020-07-06 05:12:39,587 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/8753c5df-f541-4e15-958b-713ba4d5a941/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-07 05:11:52,067 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-07 05:11:52,515 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-07 05:11:52,517 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-07 05:11:52,520 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-07 05:11:52,523 autosynth [DEBUG] > Running: git config push.default simple
2020-07-07 05:11:52,526 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-07 05:11:52,529 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-07 05:11:52,766 autosynth [INFO] > Running synthtool
2020-07-07 05:11:52,767 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-07 05:11:52,767 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-07 05:11:52,769 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-07 05:11:52,975 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-07 05:11:54,765 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmprzkgg180/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmprzkgg180/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:25.003 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-07 05:12:25,531 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmprzkgg180/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmprzkgg180/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-07 05:12:25,854 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmprzkgg180/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-07 05:12:25,879 autosynth [ERROR] > Synthesis failed
2020-07-07 05:12:25,879 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/e7c589b2-95ca-4b40-9a9a-94b15ad8fd80/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-08 05:11:56,114 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-08 05:11:56,551 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-08 05:11:56,554 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-08 05:11:56,557 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-08 05:11:56,560 autosynth [DEBUG] > Running: git config push.default simple
2020-07-08 05:11:56,562 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-08 05:11:56,565 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-08 05:11:56,803 autosynth [INFO] > Running synthtool
2020-07-08 05:11:56,803 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-08 05:11:56,803 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-08 05:11:56,805 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-08 05:11:57,019 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-08 05:11:58,871 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpr_3r9ob8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpr_3r9ob8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:29.003 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-08 05:12:29,560 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpr_3r9ob8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpr_3r9ob8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-08 05:12:29,914 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpr_3r9ob8/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-08 05:12:29,941 autosynth [ERROR] > Synthesis failed
2020-07-08 05:12:29,941 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a027909d-3dcb-472b-b848-97ccf236f75f/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-09 05:11:24,746 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-09 05:11:25,557 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-09 05:11:25,560 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-09 05:11:25,562 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-09 05:11:25,565 autosynth [DEBUG] > Running: git config push.default simple
2020-07-09 05:11:25,567 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-09 05:11:25,570 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-09 05:11:25,937 autosynth [INFO] > Running synthtool
2020-07-09 05:11:25,938 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-09 05:11:25,938 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-09 05:11:25,940 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-09 05:11:26,127 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-09 05:11:27,505 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpth6qqp15/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpth6qqp15/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:55.884 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-09 05:11:56,397 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpth6qqp15/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpth6qqp15/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-09 05:11:56,675 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpth6qqp15/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-09 05:11:56,701 autosynth [ERROR] > Synthesis failed
2020-07-09 05:11:56,702 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/2e8b015b-a1db-45fa-9bc3-8b25898b9e16/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-10 05:11:11,551 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-10 05:11:12,001 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-10 05:11:12,004 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-10 05:11:12,007 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-10 05:11:12,010 autosynth [DEBUG] > Running: git config push.default simple
2020-07-10 05:11:12,013 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-10 05:11:12,016 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-10 05:11:12,442 autosynth [INFO] > Running synthtool
2020-07-10 05:11:12,443 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-10 05:11:12,443 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-10 05:11:12,445 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-10 05:11:12,646 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-10 05:11:14,146 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpd_3yd_xm/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpd_3yd_xm/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:44.136 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-10 05:11:44,666 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpd_3yd_xm/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpd_3yd_xm/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-10 05:11:44,959 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpd_3yd_xm/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-10 05:11:44,987 autosynth [ERROR] > Synthesis failed
2020-07-10 05:11:44,988 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/9786a1a9-4acd-4b7a-98cb-0dc8104cbbbe/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-11 05:09:51,321 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-11 05:09:51,756 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-11 05:09:51,759 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-11 05:09:51,762 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-11 05:09:51,764 autosynth [DEBUG] > Running: git config push.default simple
2020-07-11 05:09:51,767 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-11 05:09:51,770 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-11 05:09:51,990 autosynth [INFO] > Running synthtool
2020-07-11 05:09:51,990 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-11 05:09:51,990 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-11 05:09:51,992 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-11 05:09:52,180 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-11 05:09:53,571 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpw_ox2i2a/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpw_ox2i2a/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:10:23.412 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-11 05:10:23,938 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpw_ox2i2a/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpw_ox2i2a/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-11 05:10:24,237 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpw_ox2i2a/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-11 05:10:24,263 autosynth [ERROR] > Synthesis failed
2020-07-11 05:10:24,263 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/746d36cc-3105-4c1c-8239-d21a57506664/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-12 05:10:33,637 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-12 05:10:34,492 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-12 05:10:34,495 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-12 05:10:34,499 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-12 05:10:34,502 autosynth [DEBUG] > Running: git config push.default simple
2020-07-12 05:10:34,505 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-12 05:10:34,508 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-12 05:10:34,779 autosynth [INFO] > Running synthtool
2020-07-12 05:10:34,779 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-12 05:10:34,779 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-12 05:10:34,782 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-12 05:10:34,984 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-12 05:10:36,452 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpjszc3709/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpjszc3709/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:07.252 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-12 05:11:07,794 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpjszc3709/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpjszc3709/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-12 05:11:08,090 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpjszc3709/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-12 05:11:08,119 autosynth [ERROR] > Synthesis failed
2020-07-12 05:11:08,119 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/7a5c4574-31e8-4eeb-ac3d-21d5ff1a628f/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-13 05:10:19,076 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-13 05:10:19,601 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-13 05:10:19,605 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-13 05:10:19,608 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-13 05:10:19,610 autosynth [DEBUG] > Running: git config push.default simple
2020-07-13 05:10:19,613 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-13 05:10:19,616 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-13 05:10:19,852 autosynth [INFO] > Running synthtool
2020-07-13 05:10:19,852 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-13 05:10:19,852 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-13 05:10:19,854 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-13 05:10:20,049 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-13 05:10:21,453 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpbjeuuaox/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpbjeuuaox/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:10:51.286 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-13 05:10:51,804 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpbjeuuaox/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpbjeuuaox/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-13 05:10:52,084 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpbjeuuaox/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-13 05:10:52,112 autosynth [ERROR] > Synthesis failed
2020-07-13 05:10:52,112 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/c317fa59-1797-4aa4-afbe-7453f2d2c6f4/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-14 05:10:24,485 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-14 05:10:25,337 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-14 05:10:25,340 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-14 05:10:25,343 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-14 05:10:25,345 autosynth [DEBUG] > Running: git config push.default simple
2020-07-14 05:10:25,348 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-14 05:10:25,351 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-14 05:10:25,567 autosynth [INFO] > Running synthtool
2020-07-14 05:10:25,568 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-14 05:10:25,568 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-14 05:10:25,570 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-14 05:10:25,768 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-14 05:10:27,199 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpriyb5oz7/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpriyb5oz7/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:10:56.970 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-14 05:10:57,495 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpriyb5oz7/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpriyb5oz7/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-14 05:10:57,783 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpriyb5oz7/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-14 05:10:57,810 autosynth [ERROR] > Synthesis failed
2020-07-14 05:10:57,811 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/0e001bf2-37c8-4b2a-9581-2b2da8a55082/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-15 05:10:55,373 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-15 05:10:55,818 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-15 05:10:55,821 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-15 05:10:55,824 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-15 05:10:55,827 autosynth [DEBUG] > Running: git config push.default simple
2020-07-15 05:10:55,830 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-15 05:10:55,833 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-15 05:10:56,094 autosynth [INFO] > Running synthtool
2020-07-15 05:10:56,094 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-15 05:10:56,095 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-15 05:10:56,097 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-15 05:10:56,314 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-15 05:10:58,281 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpekfd77ed/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpekfd77ed/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:29.770 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-15 05:11:30,491 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpekfd77ed/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpekfd77ed/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-15 05:11:30,884 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpekfd77ed/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-15 05:11:30,910 autosynth [ERROR] > Synthesis failed
2020-07-15 05:11:30,910 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/c9a16592-74a8-48fa-b603-4fc341a93800/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-16 05:12:31,292 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-16 05:12:31,738 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-16 05:12:31,741 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-16 05:12:31,744 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-16 05:12:31,747 autosynth [DEBUG] > Running: git config push.default simple
2020-07-16 05:12:31,750 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-16 05:12:31,753 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-16 05:12:32,128 autosynth [INFO] > Running synthtool
2020-07-16 05:12:32,129 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-16 05:12:32,129 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-16 05:12:32,131 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-16 05:12:32,356 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-16 05:12:34,147 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpdcmozzx_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpdcmozzx_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:13:05.925 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-16 05:13:06,499 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpdcmozzx_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpdcmozzx_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-16 05:13:06,866 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpdcmozzx_/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-16 05:13:06,895 autosynth [ERROR] > Synthesis failed
2020-07-16 05:13:06,896 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/3d2c5672-5d24-493e-860a-b190a3995941/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-17 05:11:08,553 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-17 05:11:08,997 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-17 05:11:09,000 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-17 05:11:09,003 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-17 05:11:09,005 autosynth [DEBUG] > Running: git config push.default simple
2020-07-17 05:11:09,008 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-17 05:11:09,011 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-17 05:11:09,258 autosynth [INFO] > Running synthtool
2020-07-17 05:11:09,258 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-17 05:11:09,259 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-17 05:11:09,261 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-17 05:11:09,456 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-17 05:11:10,893 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp05fzzmzv/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp05fzzmzv/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:41.684 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-17 05:11:42,228 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp05fzzmzv/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp05fzzmzv/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-17 05:11:42,540 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp05fzzmzv/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-17 05:11:42,568 autosynth [ERROR] > Synthesis failed
2020-07-17 05:11:42,569 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b06c55e1-9811-4cdf-ad4a-edf88bf922d8/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-18 05:12:04,831 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-18 05:12:05,272 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-18 05:12:05,275 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-18 05:12:05,278 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-18 05:12:05,281 autosynth [DEBUG] > Running: git config push.default simple
2020-07-18 05:12:05,284 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-18 05:12:05,287 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-18 05:12:05,653 autosynth [INFO] > Running synthtool
2020-07-18 05:12:05,653 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-18 05:12:05,653 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-18 05:12:05,656 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-18 05:12:05,872 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-18 05:12:07,696 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpgkodzd0f/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpgkodzd0f/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:39.099 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-18 05:12:39,624 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpgkodzd0f/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpgkodzd0f/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-18 05:12:39,998 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpgkodzd0f/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-18 05:12:40,026 autosynth [ERROR] > Synthesis failed
2020-07-18 05:12:40,026 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/c32ffdc4-8c58-49c9-939f-debd2b7fab92/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-19 05:12:19,552 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-19 05:12:20,404 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-19 05:12:20,407 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-19 05:12:20,410 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-19 05:12:20,413 autosynth [DEBUG] > Running: git config push.default simple
2020-07-19 05:12:20,416 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-19 05:12:20,419 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-19 05:12:20,798 autosynth [INFO] > Running synthtool
2020-07-19 05:12:20,798 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-19 05:12:20,798 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-19 05:12:20,800 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-19 05:12:21,026 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-19 05:12:22,874 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpj19knxt_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpj19knxt_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:55.193 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-19 05:12:55,755 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpj19knxt_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpj19knxt_/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-19 05:12:56,141 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpj19knxt_/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-19 05:12:56,170 autosynth [ERROR] > Synthesis failed
2020-07-19 05:12:56,170 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/97ef7833-b5ac-49c5-82b7-543ef7e5198e/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-20 05:12:35,586 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-20 05:12:36,035 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-20 05:12:36,039 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-20 05:12:36,042 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-20 05:12:36,045 autosynth [DEBUG] > Running: git config push.default simple
2020-07-20 05:12:36,049 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-20 05:12:36,052 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-20 05:12:36,276 autosynth [INFO] > Running synthtool
2020-07-20 05:12:36,276 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-20 05:12:36,276 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-20 05:12:36,278 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-20 05:12:36,483 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-20 05:12:37,953 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpc9tt2pc5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpc9tt2pc5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:13:10.220 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-20 05:13:10,980 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpc9tt2pc5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpc9tt2pc5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-20 05:13:11,277 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpc9tt2pc5/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-20 05:13:11,310 autosynth [ERROR] > Synthesis failed
2020-07-20 05:13:11,310 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/5ee7fb60-23e9-402e-adc0-551be75906b3/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-22 05:12:48,417 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-22 05:12:48,893 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-22 05:12:48,896 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-22 05:12:48,900 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-22 05:12:48,904 autosynth [DEBUG] > Running: git config push.default simple
2020-07-22 05:12:48,908 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-22 05:12:48,912 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-22 05:12:49,174 autosynth [INFO] > Running synthtool
2020-07-22 05:12:49,174 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-22 05:12:49,174 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-22 05:12:49,177 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-22 05:12:49,419 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-22 05:12:51,314 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpewpi1djf/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpewpi1djf/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:13:23.246 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-22 05:13:23,787 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpewpi1djf/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpewpi1djf/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-22 05:13:24,164 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpewpi1djf/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-22 05:13:24,193 autosynth [ERROR] > Synthesis failed
2020-07-22 05:13:24,194 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/47f45c5b-3d30-4347-b7a7-44f8ca447132/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-23 05:11:58,853 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-23 05:11:59,332 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-23 05:11:59,335 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-23 05:11:59,339 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-23 05:11:59,342 autosynth [DEBUG] > Running: git config push.default simple
2020-07-23 05:11:59,346 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-23 05:11:59,350 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-23 05:11:59,625 autosynth [INFO] > Running synthtool
2020-07-23 05:11:59,625 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-23 05:11:59,625 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-23 05:11:59,628 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-23 05:11:59,865 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-23 05:12:01,395 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmplb24neci/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmplb24neci/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:32.406 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-23 05:12:32,967 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmplb24neci/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmplb24neci/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-23 05:12:33,336 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmplb24neci/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-23 05:12:33,364 autosynth [ERROR] > Synthesis failed
2020-07-23 05:12:33,364 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/60523d84-926b-4c96-a745-3a707c5c925d/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-24 05:11:26,963 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-24 05:11:27,394 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-24 05:11:27,397 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-24 05:11:27,400 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-24 05:11:27,403 autosynth [DEBUG] > Running: git config push.default simple
2020-07-24 05:11:27,405 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-24 05:11:27,408 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-24 05:11:27,805 autosynth [INFO] > Running synthtool
2020-07-24 05:11:27,805 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-24 05:11:27,805 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-24 05:11:27,808 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-24 05:11:28,012 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-24 05:11:29,430 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp7hkius84/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp7hkius84/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:58.610 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-24 05:11:59,153 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp7hkius84/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp7hkius84/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-24 05:11:59,449 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp7hkius84/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-24 05:11:59,479 autosynth [ERROR] > Synthesis failed
2020-07-24 05:11:59,480 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/22a6a9b1-56fe-4959-b756-bec3bedf25bf/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-25 05:12:30,159 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-25 05:12:30,985 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-25 05:12:30,988 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-25 05:12:30,991 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-25 05:12:30,994 autosynth [DEBUG] > Running: git config push.default simple
2020-07-25 05:12:30,997 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-25 05:12:31,000 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-25 05:12:31,222 autosynth [INFO] > Running synthtool
2020-07-25 05:12:31,222 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-25 05:12:31,223 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-25 05:12:31,225 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-25 05:12:31,459 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-25 05:12:33,319 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp5qg4h_uw/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp5qg4h_uw/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:13:03.962 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-25 05:13:04,499 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp5qg4h_uw/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp5qg4h_uw/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-25 05:13:04,897 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp5qg4h_uw/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-25 05:13:04,922 autosynth [ERROR] > Synthesis failed
2020-07-25 05:13:04,923 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/cbb07969-162f-4ce0-83d4-4b2bdb8b201f/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-26 05:11:37,804 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-26 05:11:38,262 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-26 05:11:38,265 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-26 05:11:38,268 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-26 05:11:38,271 autosynth [DEBUG] > Running: git config push.default simple
2020-07-26 05:11:38,274 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-26 05:11:38,277 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-26 05:11:38,528 autosynth [INFO] > Running synthtool
2020-07-26 05:11:38,528 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-26 05:11:38,529 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-26 05:11:38,536 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-26 05:11:38,749 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-26 05:11:40,200 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp5g3q5sn6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp5g3q5sn6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:10.410 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-26 05:12:10,966 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp5g3q5sn6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp5g3q5sn6/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-26 05:12:11,281 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp5g3q5sn6/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-26 05:12:11,312 autosynth [ERROR] > Synthesis failed
2020-07-26 05:12:11,312 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/d2387440-d701-49b6-bf87-ae498466301f/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-27 05:12:09,185 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-27 05:12:10,031 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-27 05:12:10,034 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-27 05:12:10,037 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-27 05:12:10,039 autosynth [DEBUG] > Running: git config push.default simple
2020-07-27 05:12:10,042 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-27 05:12:10,045 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-27 05:12:10,292 autosynth [INFO] > Running synthtool
2020-07-27 05:12:10,292 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-27 05:12:10,292 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-27 05:12:10,294 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-27 05:12:10,528 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-27 05:12:12,434 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp5jy8ju1x/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp5jy8ju1x/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:42.494 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-27 05:12:43,015 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp5jy8ju1x/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp5jy8ju1x/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-27 05:12:43,370 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp5jy8ju1x/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-27 05:12:43,397 autosynth [ERROR] > Synthesis failed
2020-07-27 05:12:43,398 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/360559af-c4f0-41c3-835f-9202651ccd6e/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-27 18:21:49,211 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-27 18:21:49,649 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-27 18:21:49,652 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-27 18:21:49,654 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-27 18:21:49,657 autosynth [DEBUG] > Running: git config push.default simple
2020-07-27 18:21:49,659 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-27 18:21:49,662 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-27 18:21:50,056 autosynth [INFO] > Running synthtool
2020-07-27 18:21:50,056 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-27 18:21:50,056 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-27 18:21:50,058 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-27 18:21:50,262 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-27 18:21:51,669 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp8_h64jwy/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp8_h64jwy/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

01:22:21.540 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-27 18:22:22,246 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp8_h64jwy/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp8_h64jwy/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-27 18:22:22,543 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp8_h64jwy/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-27 18:22:22,570 autosynth [ERROR] > Synthesis failed
2020-07-27 18:22:22,571 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/c4e44801-8884-4c1f-ace9-249d6260d7a5/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-28 05:11:28,224 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-28 05:11:29,058 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-28 05:11:29,060 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-28 05:11:29,063 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-28 05:11:29,066 autosynth [DEBUG] > Running: git config push.default simple
2020-07-28 05:11:29,069 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-28 05:11:29,072 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-28 05:11:29,303 autosynth [INFO] > Running synthtool
2020-07-28 05:11:29,303 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-28 05:11:29,303 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-28 05:11:29,305 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-28 05:11:29,516 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-28 05:11:30,980 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpusle31s5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpusle31s5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:00.413 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-28 05:12:00,957 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpusle31s5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpusle31s5/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-28 05:12:01,304 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpusle31s5/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-28 05:12:01,332 autosynth [ERROR] > Synthesis failed
2020-07-28 05:12:01,333 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/d914b0a0-f36e-46d0-8d09-cbe9e662af84/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-29 05:10:59,975 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-29 05:11:00,809 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-29 05:11:00,812 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-29 05:11:00,814 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-29 05:11:00,817 autosynth [DEBUG] > Running: git config push.default simple
2020-07-29 05:11:00,820 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-29 05:11:00,823 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-29 05:11:01,097 autosynth [INFO] > Running synthtool
2020-07-29 05:11:01,097 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-29 05:11:01,097 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-29 05:11:01,099 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-29 05:11:01,302 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-29 05:11:02,771 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpw_5rbza8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpw_5rbza8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:11:32.451 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-29 05:11:32,992 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpw_5rbza8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpw_5rbza8/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-29 05:11:33,299 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpw_5rbza8/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-29 05:11:33,325 autosynth [ERROR] > Synthesis failed
2020-07-29 05:11:33,325 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a6102658-135e-4c8f-b89a-1c5d43f9b964/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-29 11:58:41,037 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-29 11:58:41,486 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-29 11:58:41,489 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-29 11:58:41,492 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-29 11:58:41,495 autosynth [DEBUG] > Running: git config push.default simple
2020-07-29 11:58:41,498 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-29 11:58:41,501 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-29 11:58:41,873 autosynth [INFO] > Running synthtool
2020-07-29 11:58:41,873 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-29 11:58:41,873 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-29 11:58:41,875 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-29 11:58:42,083 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-29 11:58:43,567 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmpvnxf0h1k/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmpvnxf0h1k/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

18:59:15.218 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-29 11:59:15,736 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmpvnxf0h1k/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmpvnxf0h1k/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-29 11:59:16,072 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmpvnxf0h1k/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-29 11:59:16,102 autosynth [ERROR] > Synthesis failed
2020-07-29 11:59:16,103 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/c5cc4ad0-adfe-40e8-a2f1-0edbd15f03ce/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

Autosynth is still having trouble generating AndroidPublisher. :sob:

Here's the output from running `synth.py`:

```
2020-07-30 05:11:44,531 autosynth [INFO] > logs will be written to: /tmpfs/src/logs/elixir-google-api
2020-07-30 05:11:44,963 autosynth [DEBUG] > Running: git config --global core.excludesfile /home/kbuilder/.autosynth-gitignore
2020-07-30 05:11:44,965 autosynth [DEBUG] > Running: git config user.name yoshi-automation
2020-07-30 05:11:44,968 autosynth [DEBUG] > Running: git config user.email yoshi-automation@google.com
2020-07-30 05:11:44,970 autosynth [DEBUG] > Running: git config push.default simple
2020-07-30 05:11:44,973 autosynth [DEBUG] > Running: git branch -f autosynth-androidpublisher
2020-07-30 05:11:44,976 autosynth [DEBUG] > Running: git checkout autosynth-androidpublisher
Switched to branch 'autosynth-androidpublisher'
2020-07-30 05:11:45,394 autosynth [INFO] > Running synthtool
2020-07-30 05:11:45,394 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--']
2020-07-30 05:11:45,394 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/elixir-google-api/AndroidPublisher/sponge_log.log
2020-07-30 05:11:45,396 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata clients/android_publisher/synth.metadata synth.py -- AndroidPublisher
2020-07-30 05:11:45,612 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/elixir-google-api/synth.py.
On branch autosynth-androidpublisher
nothing to commit, working tree clean
2020-07-30 05:11:47,425 synthtool [DEBUG] > Running: docker run --rm -v/tmpfs/tmp/tmp1mf21l8e/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
DEBUG:synthtool:Running: docker run --rm -v/tmpfs/tmp/tmp1mf21l8e/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher
/workspace /workspace
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  certifi 2.5.1
  google_api_discovery 0.7.0
  google_gax 0.3.2
  hackney 1.15.2
  idna 6.0.0
  jason 1.2.1
  metrics 1.0.1
  mime 1.3.1
  mimerl 1.2.0
  oauth2 0.9.4
  parse_trans 3.3.0
  poison 3.1.0
  ssl_verify_fun 1.1.5
  temp 0.4.7
  tesla 1.3.3
  unicode_util_compat 0.4.1
* Getting google_api_discovery (Hex package)
* Getting tesla (Hex package)
* Getting oauth2 (Hex package)
* Getting temp (Hex package)
* Getting jason (Hex package)
* Getting poison (Hex package)
* Getting hackney (Hex package)
* Getting certifi (Hex package)
* Getting idna (Hex package)
* Getting metrics (Hex package)
* Getting mimerl (Hex package)
* Getting ssl_verify_fun (Hex package)
* Getting unicode_util_compat (Hex package)
* Getting parse_trans (Hex package)
* Getting mime (Hex package)
* Getting google_gax (Hex package)
[33mThe mix.lock file was generated with a newer version of Hex. Update your client by running `mix local.hex` to avoid losing data.[0m
==> temp
Compiling 3 files (.ex)
Generated temp app
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> jason
Compiling 8 files (.ex)
Generated jason app
warning: String.strip/1 is deprecated. Use String.trim/1 instead
  /workspace/deps/poison/mix.exs:4

==> poison
Compiling 4 files (.ex)
warning: Integer.to_char_list/2 is deprecated. Use Integer.to_charlist/2 instead
  lib/poison/encoder.ex:173

Generated poison app
==> ssl_verify_fun
Compiling 7 files (.erl)
Generated ssl_verify_fun app
===> Compiling certifi
===> Compiling hackney
==> oauth2
Compiling 13 files (.ex)
Generated oauth2 app
==> mime
Compiling 2 files (.ex)
Generated mime app
==> tesla
Compiling 26 files (.ex)
Generated tesla app
==> google_gax
Compiling 5 files (.ex)
Generated google_gax app
==> google_api_discovery
Compiling 21 files (.ex)
Generated google_api_discovery app
==> google_apis
Compiling 27 files (.ex)
warning: System.cwd/0 is deprecated. Use File.cwd/0 instead
  lib/google_apis/publisher.ex:24

Generated google_apis app

12:12:16.133 [info]  FETCHING: https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest
{:error, "Error received status: 404 from discovery endpoint"}
** (File.Error) could not read file "/workspace/specifications/gdd/AndroidPublisher-v2.json": no such file or directory
    (elixir) lib/file.ex:353: File.read!/1
    lib/google_apis/generator/elixir_generator/token.ex:79: GoogleApis.Generator.ElixirGenerator.Token.build/1
    lib/google_apis/generator/elixir_generator.ex:40: GoogleApis.Generator.ElixirGenerator.generate_client/1
    lib/mix/tasks/google_apis.generate.ex:33: anonymous fn/1 in Mix.Tasks.GoogleApis.Generate.builder/1
    (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:783: Enum.each/2
    lib/mix/tasks/google_apis.generate.ex:31: Mix.Tasks.GoogleApis.Generate.builder/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
fixing file permissions
2020-07-30 05:12:16,683 synthtool [ERROR] > Failed executing docker run --rm -v/tmpfs/tmp/tmp1mf21l8e/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
ERROR:synthtool:Failed executing docker run --rm -v/tmpfs/tmp/tmp1mf21l8e/repo:/workspace -v/var/run/docker.sock:/var/run/docker.sock -e USER_GROUP=1000:1000 -w /workspace gcr.io/cloud-devrel-public-resources/elixir19 scripts/generate_client.sh AndroidPublisher:

None
2020-07-30 05:12:17,014 synthtool [DEBUG] > Wrote metadata to clients/android_publisher/synth.metadata.
DEBUG:synthtool:Wrote metadata to clients/android_publisher/synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/elixir-google-api/synth.py", line 59, in <module>
    shell.run(command, cwd=repository, hide_output=False)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-v/tmpfs/tmp/tmp1mf21l8e/repo:/workspace', '-v/var/run/docker.sock:/var/run/docker.sock', '-e', 'USER_GROUP=1000:1000', '-w', '/workspace', 'gcr.io/cloud-devrel-public-resources/elixir19', 'scripts/generate_client.sh', 'AndroidPublisher']' returned non-zero exit status 1.
2020-07-30 05:12:17,040 autosynth [ERROR] > Synthesis failed
2020-07-30 05:12:17,040 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 597, in _inner_main
    ).synthesize(synth_log_path / "sponge_log.log")
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'clients/android_publisher/synth.metadata', 'synth.py', '--', 'AndroidPublisher']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/98c6d328-bc94-423b-a588-d635683361bf/targets/github%2Fsynthtool;config=default/tests;query=elixir-google-api;failed=false).

--- a/config/apis.json
+++ b/config/apis.json
@@ -100,11 +100,6 @@
     "name": "AndroidManagement"
   },
   {
-    "version": "v2",
-    "url": "https://www.googleapis.com/discovery/v1/apis/androidpublisher/v2/rest",
-    "name": "AndroidPublisher"
-  },
-  {
     "version": "v3",
     "url": "https://www.googleapis.com/discovery/v1/apis/androidpublisher/v3/rest",
     "name": "AndroidPublisher"

